server/schema/user/user.resolver.js

```diff
@@ -102,6 +102,18 @@ module.exports = {
       throw APIError(null, error);
     }
   },
+
+  getUserByOldUserName: async (_parent, { oldUserName }, { API: { User } }) => {
+    try {
+      const _userId = await User.getIdByOldUserName(oldUserName);
+
+      const user = await User.findByID.load(_userId);
```

Comment:
There is no need for this fetch to take place twice. Just get the full user in datasources and prime it by ID.
See article datasources for reference.